### PR TITLE
feat: send funds

### PR DESCRIPTION
**Pull request by tomaszantas**
Description
---

- [x] send form 
- [x] adjust style of wallet box

Next PRs:
- wire with the backend (#355)
- Improve UI of Send Modal (Amount Input, multiline address, TBot radar background shadow) (#354)

Motivation and Context
---

#18 

How Has This Been Tested?
---

image

https://user-images.githubusercontent.com/11715931/176913355-eb4b55dc-7f6a-472b-b7a4-0c220c427962.mov

### CODE REVIEW

File: applications/launchpad/gui-react/src/components/Inputs/AmountInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  fee,
+  testId,
+}: AmountInputProps) => {
+  const dispatch = useDispatch()
+  const theme = useTheme()
+
+  const [valStr, setValStr] = useState(value.toString())
+
+  const onChangeLocal = (e: ChangeEvent<HTMLInputElement>) => {
+    // Remove all non-digits and delimiter characters:
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/components/Inputs/AmountInput/index.tsx:52]: this smells, hopefully doesn't break too soon xD

File: applications/launchpad/gui-react/src/containers/WalletContainer/Send/SendModal.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    setIsProcessing(true)
+  }
+
+  const validateAmount = (amount: number) => {
+    if (amount > available + fee) {
+      return t.wallet.transaction.errors.exceedsAvailableAndFee
+    }
+
+    if (amount === 0) {
+      return ''
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/containers/WalletContainer/Send/SendModal.tsx]: I'm not sure, is there a difference in behaviour between returning `''` and `undefined` ?

File: applications/launchpad/gui-react/src/containers/WalletContainer/Send/SendModal.tsx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+            <Text
+              type='microMedium'
+              style={{ textAlign: 'center' }}
+              color={theme.secondary}
+            >
+              {t.wallet.transaction.transactionPendingDesc2}
+            </Text>
+          </ResultModalContent>
+          <CtaButtonContainer>
+            <Button onClick={() => cancel()} fullWidth>
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/containers/WalletContainer/Send/SendModal.tsx]:
```suggestion
            <Button onClick={cancel} fullWidth>
```

File: applications/launchpad/gui-react/src/containers/WalletContainer/Send/SendModal.tsx [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+              width={88}
+              height={88}
+              style={{ marginBottom: theme.spacingVertical(1.5) }}
+            />
+            <Text type='smallMedium' style={{ textAlign: 'center' }}>
+              {t.wallet.transaction.completingDescription}
+            </Text>
+          </ResultModalContent>
+          <CtaButtonContainer>
+            <Button onClick={() => cancel()} fullWidth>
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/containers/WalletContainer/Send/SendModal.tsx]:
```suggestion
            <Button onClick={cancel} fullWidth>
```

File: applications/launchpad/gui-react/src/containers/WalletContainer/Send/SendModal.tsx [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+            rules={{
+              validate: { validateAmount },
+            }}
+            render={({ field }) => (
+              <AmountInput
+                testId='send-amount-input'
+                maxDecimals={2}
+                icon={<SvgTariSignetGradient />}
+                onChange={field.onChange}
+                value={field?.value}
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/containers/WalletContainer/Send/SendModal.tsx]: you optional chain here, but not one line above, why is that?

File: applications/launchpad/gui-react/src/containers/WalletContainer/Send/styles.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
+import styled from 'styled-components'
+
+export const StyledSendForm = styled.form`
+  display: flex;
+  flex-direction: column;
+  padding: ${({ theme }) => theme.spacing(1.67)};
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/containers/WalletContainer/Send/styles.ts:6]: where do you get these values ^^

File: applications/launchpad/gui-react/src/utils/Format.test.ts [hunk 7]
```diff
@@ -48,4 +48,12 @@ describe('Format', () => {
   it('converts Micro Tauri (uT) to Tauri (T)', () => {
     expect(toMicroT(123.456789)).toBe(123456789)
   })
+
+  it('formats the 0 coin amount to 00,000', () => {
+    expect(formatAmount(0)).toBe('00,000')
+  })
+
+  it('formats the non-zero coin amount to 00,000.00', () => {
+    expect(formatAmount(123123123.789)).toBe('123,123,123.79')
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/utils/Format.test.ts]: risky test, locale dependent

**Review by tarnas14 — DISMISSED** [review]
a couple comments, non deal breakers

File: applications/launchpad/gui-react/src/utils/Format.test.ts (hunk 7 above)
Comment (tomaszantas) [reply applications/launchpad/gui-react/src/utils/Format.test.ts]: Good point! Fixed

File: applications/launchpad/gui-react/src/containers/WalletContainer/Send/styles.ts (hunk 6 above)
Comment (tomaszantas) [reply applications/launchpad/gui-react/src/containers/WalletContainer/Send/styles.ts:6]: It's a padding inside of the modal, equal to 40px. I picked this `1.67` to be close to the designs.

File: applications/launchpad/gui-react/src/components/Inputs/AmountInput/index.tsx (hunk 1 above)
Comment (tomaszantas) [reply applications/launchpad/gui-react/src/components/Inputs/AmountInput/index.tsx:52]: It was bit challenging, but should work fine. It just removes anything else than numbers and delimiter. And delimiter is detected by the locale.

**Review by tarnas14 — APPROVED** [review]

**Pull request merged**